### ISSUE TITLE
Changes in pg_dump/pg_restore 9.6 break RULE extraction

### ISSUE THREAD

**Issue by oldlance** [omniti-labs/pg_extractor#40]
_Version info_
pg_extractor.py : 2.3.7
postgresql servers : 8.4, 9.4, 9.5 & 9.6
pg_dump : 9.6.1 (until recently 9.5.5)
pg_restore : 9.6.1 (until recently 9.5.5)

First, thanks for a **great** utility which I've been using successfully for a few years to take nightly snapshots of my databases' schema DDL.  Recently I noticed some unexpected changes in the extracted DDL for my rules.  They seemed to be switching back and forth between INSERT, UPDATE and DELETE variants.

Further investigation revealed that instead of there being one file **per rule** in ./rules/, there is only one file **per view**.  So, for example, instead of seeing, as I did in the past:

```
schema_name.view_name1_update.sql
schema_name.view_name1_insert.sql
schema_name.view_name1_delete.sql

schema_name.view_name2_update.sql
schema_name.view_name2_insert.sql
schema_name.view_name2_delete.sql
```
I now see

```
schema_name.view_name1.sql
schema_name.view_name2.sql
```
The underlying cause seems to be a difference in the information stored in the dump file prepared using `pg_dump 9.6.1`.  

If I run `pg_restore --list` on a dump file generated using `pg_dump 9.5.5` using `pg_restore 9.5.5`  *or* `pg_restore 9.6.1` I see the following output:

```
nnnn; nnnn nnnnnnnnn RULE schema_name rule_name owner_name
```

If I run `pg_restore --list` on a dump file generated using `pg_dump 9.6.1` using `pg_restore 9.6.1` I see the following output:

```
nnnn; nnnn nnnnnnnnn RULE schema_name view_name rule_name owner_name
```
That is, _view_name_ has been inserted between _schema_name_ and _rule_name_ and consequently `pg_extractor.py` is extracting the wrong information when building the filename.

**Comment by keithf4** [comment]
Thanks for letting me know! Not running 9.6 very many places yet myself. Will see what I can do to fix it.

**Comment by oldlance** [comment]
Thanks for your response and apologies for not being able to submit a fix.

**Comment by keithf4** [comment]
Working on a fix for this, and also found some other things to clean up as well.

One interesting thing I found, and I'm not sure I'll easily be able to do anything about it, is it's actually the version of pg_dump that creates the binary dump file that determines whether the restore list has the view/table as part of the entry. I created a dump of a 9.5 database with the 9.6 pg_dump, and even running the 9.5 pg_restore causes the newer output in the restore list.

Don't think it really matters since pg_extractor makes its own binary schema dump file to use, but something to keep in mind I guess.

**Comment by keithf4** [comment]
Can you give the new version of the script in the Gist below a try and make sure it works for your RULEs? I tested with some basic ones and it seems fine, but would be good to know it works for yours before release

https://gist.github.com/keithf4/1599da0c0bc75543d66906be372566e9

**Comment by oldlance** [comment]
Thanks for the very quick turnaround, @keithf4!

I tested my the script in the gist against one of my databases using both `pg_dump 9.6.1` and `pg_dump 9.5.5` and the extracted rules were correct in both cases;  also, the corrections were the only changes in any of the database DDL (which is good!).

I'd say that it's fixed.

**Comment by keithf4** [comment]
Released version 2.3.8 with this fix.

Fixing this also made me realize I could simplify a good chunk of other code. So thanks :)

**Comment by oldlance** [comment]
Thanks once again, @keithf4!!  Did a production run using 2.3.8 and all is good.
